dll_injector/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+use std::{
+    mem::{size_of, transmute},
+    ptr,
+};
+
+use error::InjectorError;
+use windows::{
+    s, w,
+    Win32::{
+        Foundation::{CloseHandle, HANDLE},
+        System::{
+            Diagnostics::{
+                Debug::WriteProcessMemory,
+                ToolHelp::{
+                    CreateToolhelp32Snapshot, Process32FirstW, Process32NextW, Thread32First,
+                    Thread32Next, PROCESSENTRY32W, TH32CS_SNAPPROCESS, TH32CS_SNAPTHREAD,
+                    THREADENTRY32,
+                },
+            },
+            LibraryLoader::{GetModuleHandleW, GetProcAddress},
+            Memory::{VirtualAllocEx, MEM_COMMIT, PAGE_READWRITE},
+            Threading::{
+                CreateRemoteThread, OpenProcess, OpenThread, ResumeThread, SuspendThread,
+                PROCESS_CREATE_THREAD, PROCESS_VM_OPERATION, PROCESS_VM_WRITE, THREAD_ALL_ACCESS,
+            },
+        },
+    },
+};
+
+pub mod error;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct Process {
+    pid: u32,
+}
+
+impl Process {
+    pub fn find_process(starts_with: &str) -> Result<Option<Self>, InjectorError> {
+        let snapshot = unsafe { CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0) }?;
+
+        let mut process_entry = PROCESSENTRY32W::default();
+        process_entry.dwSize = size_of::<PROCESSENTRY32W>() as u32;
+
+        if unsafe { Process32FirstW(snapshot, &mut process_entry) }.as_bool() {
+            while unsafe { Process32NextW(snapshot, &mut process_entry) }.as_bool() {
+                let os_string = String::from_utf16(&process_entry.szExeFile[..starts_with.len()])?;
+                if os_string.to_lowercase() == *starts_with.to_lowercase() {
+                    return Ok(Some(Process {
+                        pid: process_entry.th32ProcessID,
+                    }));
+                }
+            }
+        }
+
+        Ok(None)
+    }
+
+    pub fn wait_for_process(starts_with: &str) -> Result<Self, InjectorError> {
+        let mut process = Process::find_process(starts_with)?;
+        while let None = process {
+            process = Process::find_process(starts_with)?;
+        }
+
+        Ok(process.unwrap())
```

Comment:
Maybe `.ok_or` would be more idiomatic? Depends also on Error type changes.

Reply:
There isn't a case where we get out of the loop and still have the process set to None